lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -523,26 +532,29 @@ OpFoldResult LTPrimOp::fold(ArrayRef<Attribute> operands) {
       // lt(x, const) -> 0 where const <= minValue of the unsigned type of x
       // Handled explicitly above.
 
-      // lt(x, const) -> 0 where const <= minValue of the signed type of x
-      if (!isUnsigned &&
-          rhsCst.getValue()
-              .sextOrSelf(commonWidth)
```

Comment:
can you  just add a side-function to deal with the 0-case so you don't need all the duplication?

Reply:
Yeah, I can break this out. Good idea!

Reply:
I added two support utilities, `sextOrSelfZeroWidth` and `extOrTruncZeroWidth`, that simplifies the duplication here (without simplifying the existing duplication in the code here).